nni/algorithms/compression/v2/pytorch/common/data_collector.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import logging
+from typing import Dict, List
+
+from torch import Tensor
+
+from nni.algorithms.compression.v2.pytorch.base.pruner_tools import DataCollector, TrainerBasedDataCollector
+
+_logger = logging.getLogger(__name__)
+
+__all__ = ['WeightDataCollector', 'WeightTrainerBasedDataCollector', 'SingleHookTrainerBasedDataCollector']
+
+
+class WeightDataCollector(DataCollector):
```

Comment:
Maybe `WeightDataCollector` and `TrainerBasedDataCollector` can be put into the same file since they belong to the same level?